documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+    * **_unsupported_grant_type_**
+
+      The authorization grant type is not supported by the authorization server.
+
+    * **_invalid_scope_**
+
+      The requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner.
+
+
+* **error_description**
+
+  OPTIONAL. Human-readable ASCII text providing additional information, used to assist the client developer in understanding the error that occurred. Values for the `error_description` parameter MUST NOT include characters outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+
+The parameters are included in the entity-body of the HTTP response using the `application/json` media type.
+
+`error_description` parameter is not a valid field to take decisions on the developer side, as its value may vary depending on the server implementation, language, format or context. Therefore, the developer should base their logic only on the `error` parameter.
+
+The following is a non-normative example Token Error Response:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+```
+```json
+{
+  "error": "unauthorized_client"
+}
+```
+
+## The Scope Parameter
+
+Oauth scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services.
+
+The Authorization Request MUST contain the `scope` parameter, which is a space delimited, case-sensitive list of ASCII strings (scope values). Multiple scope values can be requested simultaneously, subject to the SP being registered to use those “scopes”.
+
+For OpenID requests, the scope values MUST include `openid`, to indicate that the request is an OpenID Connect request, followed by other values depending on the specific CAMARA services being requested.
+
+A purpose must be declared within the requested scope for the following grant types:
+
+* _authorization_code_
+
+* _urn:openid:params:grant-type:ciba_
+
+No purpose is required for the `client_credentials` grant type, as this grant type can only be used when no personal user data is processed. The requested scope must be set directly to the value defined for the relevant endpoint within the OAS ("YAML") specification. "Wild card" scopes (i.e. specifying only the API name) are not valid for this grant type.
+
+To declare a purpose when accessing the CAMARA APIs, it is strongly recommended to guarantee interoperability among Authorization Servers that the `scope` parameter is set to:
+
+`dpv:<dpvValue>#<technicalParameter>`
+
+* `<dpvValue>` is coming from [W3C DPV purpose definition](https://w3c.github.io/dpv/dpv/#vocab-purpose)
+
+* `<technicalParameter>` must be either:
+
+    * one technical scope to limit the request to one API endpoint (this technical scope is described in the API spec YAML file)
+
+    * one API name to cover the complete API (This API name is not described in the API spec YAML file as technical scope) – In this case the request covers all technical scopes of the API.
```

Comment:
If we would like to continue using the wildcard scope, it should be made part of the API documentation in every subproject so that the API documentation can clearly state what it means to use a wildcard technical scope.

Reply:
I suggest to deprecate using the wildcard scope. If used at all, it must be allowed only for APIs which name it in the API documentation.

Reply:
Please see https://github.com/camaraproject/IdentityAndConsentManagement/pull/113#discussion_r1474604755

Reply:
> Please see [#113 (comment)](https://github.com/camaraproject/IdentityAndConsentManagement/pull/113#discussion_r1474604755)

@jpengar can you shortly explain what the referred comment has to do with the content of this review comments from @shilpa-padgaonkar and @Elisabeth-Ericsson here (way to handle wildcard scope?).

Reply:
> @jpengar can you shortly explain what the referred comment has to do with the content of this review comments from @shilpa-padgaonkar and @Elisabeth-Ericsson here (way to handle wildcard scope?).

@hdamker Long story short, my comment was actually more related to Eli's comment (https://github.com/camaraproject/IdentityAndConsentManagement/pull/113#discussion_r1471453854) and I just wanted to highlight why the "wildcard" scope is present in the document. My bad, I didn't make my point clear enough :/

So, my point in including that reference comment was to highlight that the reason the current profile content includes the agreed upon "wildcard" (api name, bundle scope... people call it different things xD) is because this PR proposes to use a first version of the OIDC profile as an appropriate way to cover the current state (a.k.a. short-term agreement) and urgent issues/clarifications targeted for v0.2. But of course it is still pending to discuss the new/improved solution to handle purpose and it would be properly documented in v0.3. And if decided, remove the use of wildcard scopes, etc.